resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentVidServiceImpl.java

```diff
@@ -258,7 +258,7 @@ private VidGeneratorResponseDto vidGenerator(VidRequestDto requestDto)
 		try {
 
 			// revoke vid
-			VidGeneratorResponseDto vidResponse = vidDeactivator(requestDto, uin);
+			VidGeneratorResponseDto vidResponse = vidDeactivator(requestDto, uin, vid);
```

Comment:
In the above line of 
JSONObject jsonObject = utilitiy.retrieveIdrepoJson(vid); 
can you use the requestDto.getIndividualId() instead of vid if the getIndividualType is not UIN ?

This change is required to avoid a security issue, where one can authenticate with their UIN and otp and revoke the VID of another individual.

Reply:
ok got it.